solutions/instance/main.tf

```diff
@@ -58,9 +58,13 @@ locals {
   existing_secrets_manager_instance_crn_split = var.existing_secrets_manager_instance_crn != null ? split(":", var.existing_secrets_manager_instance_crn) : null
   existing_secrets_manager_instance_guid      = var.existing_secrets_manager_instance_crn != null ? element(local.existing_secrets_manager_instance_crn_split, length(local.existing_secrets_manager_instance_crn_split) - 3) : null
   existing_secrets_manager_instance_region    = var.existing_secrets_manager_instance_crn != null ? element(local.existing_secrets_manager_instance_crn_split, length(local.existing_secrets_manager_instance_crn_split) - 5) : null
+
+  # tflint-ignore: terraform_unused_declarations
+  validate_sm_crn = length(local.service_credential_secrets) > 0 && var.existing_secrets_manager_instance_crn == null ? tobool("`var.existing_secrets_manager_instance_crn` is required when adding service credentials to a secrets manager secret.") : false
 }
 
 module "secrets_manager_service_credentials" {
+  count                       = length(local.service_credential_secrets) > 0 ? 1 : 0
```

Comment:
by adding a count, won't we need a moved block to prevent breaking change?

Reply:
In my opinion, we won't need a moved block since this feature hasn't been released due to a catalog onboarding failure.

Reply:
Someone can still consume from github or hashicorp though, so perhaps we add it anyway. There is no harm right?

Reply:
added

Reply:
Makes sense, I've added the moved block.